astropy/cosmology/core.py

```diff
@@ -393,12 +394,25 @@ def __eq__(self, other: Any, /) -> bool:
     def __repr__(self):
         namelead = f"{self.__class__.__qualname__}("
         if self.name is not None:
-            namelead += f'name="{self.name}", '
+            namelead += f"name={self.name!r}, "
         # nicely formatted parameters
-        fmtps = (f"{k}={getattr(self, k)}" for k in self.__parameters__)
+        fmtps = (f"{k}={getattr(self, k)!r}" for k in self.__parameters__)
 
         return namelead + ", ".join(fmtps) + ")"
 
+    def __str__(self):
+        """Return a string representation of the cosmology."""
+        fs = (  # name + fields
+            (
+                f"{k!s}={getattr(self, k)!s}"
+                if k != "name"
+                else f'name="{getattr(self, k)!s}"'  # name needs quotes
+            )
+            for k in chain(("name",), self.__parameters__)
```

Comment:
This code doesn't need anything from `itertools`:
```suggestion
            for k in ("name", *self.__parameters__)
```